index.js

```diff
@@ -68,20 +90,81 @@ function Chapter (index, chapterName, fileName, content) {
   return {index: index, chapterName: chapterName, fileName: fileName, content: content}
 }
 
+Streampub.prototype._generateCover = function(async) {
+  var self = this
+  var title = self.meta.title || 'Cover'
+  function execute(callback) {
+    self.write({
```

Comment:
As you've discovered ("write after end") calling "write" from the guts of a stream implementation is fraught and you shouldn't really do it.

If you want to do this calling `self._transform` would be way safer.